hls-plugin-api/src/Ide/Plugin/ConfigUtils.hs

```diff
@@ -0,0 +1,88 @@
+{-# LANGUAGE GADTs             #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards   #-}
+{-# LANGUAGE ViewPatterns      #-}
+
+module Ide.Plugin.ConfigUtils where
```

Comment:
This file is too complicated for me to understand what's going on. Some haddock + splitting out the big `where` clauses into individual functions would go a long way.